test/test-filter-process.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+. "test/testlib.sh"
+
+# HACK(taylor): git uses ".g<hash>" in the version name to signal that it is
+# from the "next" branch, which is the only (current) version of Git that has
+# support for the filter protocol.
+#
+# Once 2.11 is released, replace this with:
+#
+# ```
+# ensure_git_version_isnt $VERSION_LOWER "2.11.0"
```

Comment:
Good thought. Since adding support for the `git-describe` format would complicate this function (and I'm not planning on gating any more LFS features behind Git versions other than those released), I'm going to forgo this change for now, unless we need something like this in the future.